src/libsync/propagateuploadng.cpp

```diff
@@ -97,6 +97,12 @@ void PropagateUploadFileNG::doStartUpload()
                 this, SLOT(slotPropfindIterate(QString,QMap<QString,QString>)));
         job->start();
         return;
+    } else if (progressInfo._valid) {
+        // The upload info is stale. remove the stale chunks on the server
+        _transferId = progressInfo._transferid;
+        // Fire and forget. Any error will be ignored.
+        (new DeleteJob(propagator()->account(), chunkUrl(), this))->start();
```

Comment:
It would be a bit cleaner and safer if `chunkUrl` would take the transferId as parameter (and could be made static) instead of having to set the variable and reset it later.